### DISCUSSION TITLE
Added LifeTime production info

### DISCUSSION THREAD

**Discussion by MassimoL06** [skelgaard/homeassistant-apsystems#34] [Show and tell]
hello @skelgaard , I have added LifeTime production information in the sensor.py file.
Added this in the **SENSOR** Table
```
    SENSOR_ENERGY_TOTAL: ApsMetadata(
        json_key="lifetime",
        unit=ENERGY_KILO_WATT_HOUR,
        icon="mdi:solar-power",
        state_class="total_increasing",
    )
```
then in **class APsystemsFetcher**
I added a new url
```
url_data2 = "https://www.apsystemsema.com/ema/ajax/getDashboardApiAjax/getDashboardProductionInfoAjax"
```
then in method **async def run(self)**, I added this
```
            result_data2 = await self._hass.async_add_executor_job(
                s.request,
                "POST",
                self.url_data2,
                None,
                None,
                self.headers,
                browser.cookies.get_dict(),
            )
```
And added one JSon info into the result JSON
```
           if result_data.status_code == 204:
                self.cache = None
            else:
                self.cache = result_data.json()
                self.cache["lifetime"] = result_data2.json()["lifetime"];
```
 If you want, you can add it to your code.

Regards

**Comment by skelgaard** [comment]
if you do a PR i will merge, else i will add it my selves :)

**Reply by Udoggie** [reply]
@skelgaard  FYI, I did a pull request for this last week.

**Comment by Udoggie** [comment]
FYI, on the first change,
`   unit=ENERGY_KILO_WATT_HOUR,
should really be:
`   unit=UnitOfEnergy.KILO_WATT_HOUR,

I'll do a PR shortly.

**Comment by Udoggie** [comment]
Merged into master 0885841  (pull request #41)
This is complete
